CONTRIBUTING.md

```diff
@@ -0,0 +1,89 @@
+# Contributing to lc0
+
+These are the guidelines and standards followed by this codebase.
+
+The language is C++, specifically C++14. The compiler used must support C++14, and the entire C++14 standard featureset may be used. As such, manual `new` and `delete` memory mangement is disallowed; use the standard library tools for managing memory (such as `unique_ptr`, `shared_ptr` etc.). When compiler support is more widespread, the project may upgrade to the C++17 standard in the future.
+
+This codebase uses semantic versioning. A release is the final commit for that version number, and all subsequent commits are development for the next version. `master` is the default branch, and the active development branch (as such, all Pull Requests go here); it always targets a minor (or major) version which succeeds the current relase. `release` is always equivalent to the latest tag.
+
+
+### Style
+
+Style is of course the first guideline on every new contributor's mind :)
+
+This codebase largely complies with the [Google C++ style guide](https://google.github.io/styleguide/cppguide.html). The maintainers recommend the use of [Clang's auto formatter](https://clang.llvm.org/docs/ClangFormatStyleOptions.html).
+
+Notable exceptions:
+ 1. C++ exceptions are allowed (in fact, only `lczero::Exception`, defined in `utils/exception.h`, is allowed)
+ 2. We use `#pragma once` instead of header guards.
+ 3. Default function parameters are sometimes allowed, not only for constructors and assignment operators.
+ 4. Rvalue reference function params are sometimes allowed.
```

Comment:
Oops, I don't know what I was thinking of, but `, not only for constructors and assignment operators` actually was related to rvalue reference parameters (item 4), not default function arguments.